piptools/writer.py

```diff
@@ -264,7 +266,7 @@ def write(
             else:
                 log.info(line)
                 self.dst_file.write(unstyle(line).encode())
-                self.dst_file.write(os.linesep.encode())
+                self.dst_file.write(self.linesep.encode())
```

Comment:
I think it would be neater to use a TextIOWrapper so the existing code can continue to use `\n`, which should sharply reduce the number of lines changed here

```python
try:
    dst_file = io.TextIOWrapper(self.dst_file, encoding="utf8", newlines=self.linesep):
    for line in self._iter_lines(...):
        dst_file.write(line)
        dst_file.write("\n")
finally:
    dst_file.detatch()
```